include/matx/core/make_sparse_tensor.h

```diff
@@ -0,0 +1,139 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "matx/core/sparse_tensor.h"
+
+namespace matx {
+namespace experimental {
+
+//
+// MatX uses a single versatile sparse tensor type that uses a tensor format
+// DSL (Domain Specific Language) to describe a vast space of storage formats.
+// This file provides a number of convenience factory methods that construct
+// sparse matrices in well-known storage formats, like COO, CSR, and CSC,
+// directly from the constituent buffers. More factory methods can easily be
+// added as the need arises.
+//
+
+// Internal helper method for sparse matrices (2-dim tensors).
+template <typename VAL, typename CRD, typename POS, typename TF>
+static __MATX_INLINE__ auto
+make_sparse_matrix(const index_t (&shape)[2], VAL *val, size_t vsz, CRD *c0,
+                   size_t c0sz, CRD *c1, size_t c1sz, POS *p0, size_t p0sz,
+                   POS *p1, size_t p1sz, bool owning) {
+  // Values.
+  raw_pointer_buffer<VAL, matx_allocator<VAL>> bufv{val, vsz, owning};
+  basic_storage<decltype(bufv)> sv{std::move(bufv)};
+  // Coordinates.
+  raw_pointer_buffer<CRD, matx_allocator<CRD>> bufc0{c0, c0sz, owning};
+  raw_pointer_buffer<CRD, matx_allocator<CRD>> bufc1{c1, c1sz, owning};
+  basic_storage<decltype(bufc0)> sc0{std::move(bufc0)};
+  basic_storage<decltype(bufc1)> sc1{std::move(bufc1)};
+  // Positions.
+  raw_pointer_buffer<POS, matx_allocator<POS>> bufp0{p0, p0sz, owning};
+  raw_pointer_buffer<POS, matx_allocator<POS>> bufp1{p1, p1sz, owning};
+  basic_storage<decltype(bufp0)> sp0{std::move(bufp0)};
+  basic_storage<decltype(bufp1)> sp1{std::move(bufp1)};
+  // Sparse tensor in templated format.
+  return sparse_tensor_t<VAL, CRD, POS, TF, decltype(sv), decltype(sc0),
+                         decltype(sp0)>(shape, std::move(sv),
+                                        {std::move(sc0), std::move(sc1)},
+                                        {std::move(sp0), std::move(sp1)});
+}
+
+// Constructs a sparse matrix in COO format directly from the values and
+// coordinates arrays. The entries should be sorted by row, then column.
+// Duplicate entries should not occur. Explicit zeros may be stored.
+template <typename VAL, typename CRD>
+auto make_coo(VAL *val, CRD *row, CRD *col, const index_t (&shape)[2],
```

Comment:
trying not to be too verbose, but should this be called `make_tensor_coo` to fit with the dense naming? or maybe `make_tensor_sp_coo`?

Reply:
changed all method names accordingly